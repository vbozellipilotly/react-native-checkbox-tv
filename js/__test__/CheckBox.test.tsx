import * as React from 'react';
import renderer from 'react-test-renderer';
import AndroidCheckbox from '../CheckBox.android';
import IosCheckbox from '../CheckBox.ios';

describe('render Android <Checkbox />', () => {
  it('renders enabled Android Checkbox', () => {
    const tree = renderer.create(<AndroidCheckbox />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders disabled Android Checkbox', () => {
    const tree = renderer.create(<AndroidCheckbox disabled />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders Android Checkbox with value', () => {
    const tree = renderer.create(<AndroidCheckbox value />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('render IOS <Checkbox />', () => {
  it('renders enabled IOS Checkbox', () => {
    const tree = renderer.create(<IosCheckbox />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders hideBox IOS Checkbox', () => {
    const tree = renderer
      .create(<IosCheckbox hideBox={true} value={true} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders disabled IOS Checkbox', () => {
    const tree = renderer
      .create(<IosCheckbox disabled={true} value={true} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders disabled false IOS Checkbox', () => {
    const tree = renderer
      .create(<IosCheckbox disabled={false} value={true} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders IOS Checkbox without disabled props', () => {
    const tree = renderer.create(<IosCheckbox value={true} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders IOS Checkbox with accessible={false} props', () => {
    const tree = renderer.create(<IosCheckbox accessible={false} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders IOS Checkbox with full setting', () => {
    const tree = renderer
      .create(
        <IosCheckbox
          value={false}
          onValueChange={() => {}}
          onAnimationDidStop={() => {}}
          lineWidth={2}
          hideBox={false}
          boxType={'circle'}
          tintColor={'#9E663C'}
          onCheckColor={'#6F763F'}
          onFillColor={'#4DABEC'}
          onTintColor={'#F4DCF8'}
          animationDuration={0.5}
          disabled={false}
          onAnimationType={'bounce'}
          offAnimationType={'stroke'}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
