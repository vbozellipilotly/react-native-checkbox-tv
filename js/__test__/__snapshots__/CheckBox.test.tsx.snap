// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Android <Checkbox /> renders Android Checkbox with value 1`] = `
<AndroidCheckBox
  enabled={true}
  forwardedRef={null}
  on={true}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 32,
      },
      undefined,
    ]
  }
  value={true}
/>
`;

exports[`render Android <Checkbox /> renders disabled Android Checkbox 1`] = `
<AndroidCheckBox
  disabled={true}
  enabled={false}
  forwardedRef={null}
  on={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 32,
      },
      undefined,
    ]
  }
/>
`;

exports[`render Android <Checkbox /> renders enabled Android Checkbox 1`] = `
<AndroidCheckBox
  enabled={true}
  forwardedRef={null}
  on={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "height": 32,
        "width": 32,
      },
      undefined,
    ]
  }
/>
`;

exports[`render IOS <Checkbox /> renders IOS Checkbox with accessible={false} props 1`] = `
<View
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": false,
    }
  }
  accessibilityValue={
    Object {
      "text": "off",
    }
  }
  accessible={false}
  pointerEvents="auto"
>
  <RNCCheckbox
    forwardedRef={null}
    onValueChange={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "height": 32,
          "width": 32,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`render IOS <Checkbox /> renders IOS Checkbox with full setting 1`] = `
<View
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": false,
    }
  }
  accessibilityValue={
    Object {
      "text": "off",
    }
  }
  accessible={true}
  pointerEvents="auto"
>
  <RNCCheckbox
    animationDuration={0.5}
    boxType="circle"
    forwardedRef={null}
    hideBox={false}
    lineWidth={2}
    offAnimationType="stroke"
    onAnimationDidStop={[Function]}
    onAnimationType="bounce"
    onCheckColor="#6F763F"
    onFillColor="#4DABEC"
    onTintColor="#F4DCF8"
    onValueChange={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "height": 32,
          "width": 32,
        },
        undefined,
      ]
    }
    tintColor="#9E663C"
    value={false}
  />
</View>
`;

exports[`render IOS <Checkbox /> renders IOS Checkbox without disabled props 1`] = `
<View
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": true,
      "disabled": false,
    }
  }
  accessibilityValue={
    Object {
      "text": "on",
    }
  }
  accessible={true}
  pointerEvents="auto"
>
  <RNCCheckbox
    forwardedRef={null}
    onValueChange={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "height": 32,
          "width": 32,
        },
        undefined,
      ]
    }
    value={true}
  />
</View>
`;

exports[`render IOS <Checkbox /> renders disabled IOS Checkbox 1`] = `
<View
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": true,
      "disabled": true,
    }
  }
  accessibilityValue={
    Object {
      "text": "on",
    }
  }
  accessible={true}
  pointerEvents="none"
>
  <RNCCheckbox
    forwardedRef={null}
    onValueChange={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "height": 32,
          "width": 32,
        },
        undefined,
      ]
    }
    value={true}
  />
</View>
`;

exports[`render IOS <Checkbox /> renders disabled false IOS Checkbox 1`] = `
<View
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": true,
      "disabled": false,
    }
  }
  accessibilityValue={
    Object {
      "text": "on",
    }
  }
  accessible={true}
  pointerEvents="auto"
>
  <RNCCheckbox
    forwardedRef={null}
    onValueChange={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "height": 32,
          "width": 32,
        },
        undefined,
      ]
    }
    value={true}
  />
</View>
`;

exports[`render IOS <Checkbox /> renders enabled IOS Checkbox 1`] = `
<View
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": false,
      "disabled": false,
    }
  }
  accessibilityValue={
    Object {
      "text": "off",
    }
  }
  accessible={true}
  pointerEvents="auto"
>
  <RNCCheckbox
    forwardedRef={null}
    onValueChange={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "height": 32,
          "width": 32,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`render IOS <Checkbox /> renders hideBox IOS Checkbox 1`] = `
<View
  accessibilityRole="checkbox"
  accessibilityState={
    Object {
      "checked": true,
      "disabled": false,
    }
  }
  accessibilityValue={
    Object {
      "text": "on",
    }
  }
  accessible={true}
  pointerEvents="auto"
>
  <RNCCheckbox
    forwardedRef={null}
    hideBox={true}
    onValueChange={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "height": 32,
          "width": 32,
        },
        undefined,
      ]
    }
    value={true}
  />
</View>
`;
